odl/tomo/geometry/conebeam.py

```diff
@@ -16,13 +16,457 @@
+
+        For an angle ``phi``, the detector position is given by ::
+
+            det_ref(phi) = translation +
+                           rot_matrix(phi) * (det_rad * src_to_det_init)
+
+        where ``src_to_det_init`` is the initial unit vector pointing
+        from source to detector.
+
+        Parameters
+        ----------
+        angle : float
+            Rotation angle given in radians, must be contained in
+            this geometry's `motion_params`
+
+        Returns
+        -------
+        point : `numpy.ndarray`, shape (2,)
+            Detector reference point corresponding to the given angle
+
+        See Also
+        --------
+        rotation_matrix
+
+        Examples
+        --------
+        With default arguments, the detector starts at ``det_rad * e_y``
+        and rotates to ``det_rad * (-e_x)`` at 90 degrees:
+
+        >>> apart = odl.uniform_partition(0, 2 * np.pi, 10)
+        >>> dpart = odl.uniform_partition(-1, 1, 20)
+        >>> geom = FanFlatGeometry(apart, dpart, src_radius=2, det_radius=5)
+        >>> geom.det_refpoint(0)
+        array([ 0.,  5.])
+        >>> np.allclose(geom.det_refpoint(np.pi / 2), [-5, 0])
+        True
+        """
+        if self.check_bounds and not self.motion_params.contains_all(angle):
+            raise ValueError('`angle` {} not in the valid range {}'
+                             ''.format(angle, self.motion_params))
+
+        angle = np.array(angle, dtype=float, copy=False, ndmin=1)
+        # Initial vector from the rotation center to the detector. It can be
+        # computed this way since source and detector are at maximum distance,
+        # i.e. the connecting line passes the origin.
+        center_to_det_init = self.det_radius * self.src_to_det_init
+        refpt = (self.translation[None, :] +
+                 self.rotation_matrix(angle).dot(center_to_det_init))
+        return refpt.squeeze()
+
+    def rotation_matrix(self, angle):
+        """Return the rotation matrix for ``angle``.
+
+        For an angle ``phi``, the matrix is given by ::
+
+            rot(phi) = [[cos(phi), -sin(phi)],
+                        [sin(phi), cos(phi)]]
+
+        Parameters
+        ----------
+        angle : float
+            Rotation angle given in radians, must be contained in
+            this geometry's `motion_params`.
+
+        Returns
+        -------
+        rot : `numpy.ndarray`, shape (2, 2)
+            The rotation matrix mapping the standard basis vectors in
+            the fixed ("lab") coordinate system to the basis vectors of
+            the local coordinate system of the detector reference point,
+            expressed in the fixed system.
+        """
+        if self.check_bounds and not self.motion_params.contains_all(angle):
+            raise ValueError('`angle` {} not in the valid range {}'
+                             ''.format(angle, self.motion_params))
+
+        return euler_matrix(angle)
```

Comment:
That's just the name of the helper to compute a rotation with Euler angles. Perhaps it's a bad name in 2D..